pkg/controller/secret/secret_controller.go

```diff
@@ -0,0 +1,145 @@
+package secret
+
+import (
+	"context"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/lib/secrets"
+	"github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "controller_secret"
+	userDataSecret = "windows-user-data"
+)
+
+var log = logf.Log.WithName(ControllerName)
+
+// Add creates a new Secret Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, _ string) error {
+	reconciler, err := newReconciler(mgr)
+	if err != nil {
+		return errors.Wrapf(err, "could not create %s reconciler", ControllerName)
+	}
+	return add(mgr, reconciler)
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) (reconcile.Reconciler, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
+	if err != nil {
+		return nil, err
+	}
+
+	reconciler := &ReconcileSecret{client: client, scheme: mgr.GetScheme()}
+
+	// Create userData secret
+	userData, _ := secrets.GenerateUserData()
```

Comment:
You don't want to catch the error or does this mean, we don't care if there is userData secret creation in here as the subsequent reconciler would take care of it?